### ISSUE TITLE
Error in request to /syscheck/:agent-id with special characters in the parameter q

### ISSUE THREAD

**Issue by jsanchez91** [wazuh/wazuh#4872]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.12.x | API | Manager | --- | --- |

Hello Team!

I tried to filter the output of `/syscheck/:agent-id` by the name of the file. On Windows the file name can contain these special characters: `\`, `[`, `]`, `{`, `}`

These are some examples of the values the file field can have:

```js
GET /syscheck/:agent-id{
  "q": "file=c:\windows\sysnative\wbem\wmic.exe"
}

GET /syscheck/:agent-id{
  "q": "file=c:\windows\system32\winrm.vbs"
}

GET /syscheck/:agent-id{
  "q": "file=c:\windows\system32\wbem\wmic.exe"
}
GET /syscheck/:agent-id{
  "q": "file=c:\windows\system32\winrm.vbs;file=c:\windows\system32\wbem\wmic.exe"
}

GET /syscheck/:agent-id{
  "q": "file=[x32] HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\amdgpio2"
}

GET /syscheck/:agent-id{
  "q": "file=[x32] HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\amdi2c"
}

GET /syscheck/:agent-id{
  "q": "file=[x32] HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\RemoteAccess\Accounting\Providers\{1AA7F840-C7F5-11D0-A376-00C04FC9DA04}"
}

GET /syscheck/:agent-id{
  "q": "file=[x32] HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\RemoteAccess\Authentication\Providers\{1AA7F83F-C7F5-11D0-A376-00C04FC9DA04}"
}

GET /syscheck/:agent-id{
  "q": "file=[x32] HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\RemoteAccess\Accounting\Providers\{1AA7F840-C7F5-11D0-A376-00C04FC9DA04};file=[x32] HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\RemoteAccess\Authentication\Providers\{1AA7F83F-C7F5-11D0-A376-00C04FC9DA04}"
}
```

I would need those filters to be able to apply.
Thanks.

**Comment by davidjiglesias** [comment]
Example responses after fix:
```
root@7429ca762e39:/wazuh-api/test# curl -u foo:bar -k -X GET "https://127.0.0.1:55000/syscheck/000?pretty&q=file=c:\windows\sysnative\wbem\wmic.exe"
{
   "error": 0,
   "data": {
      "items": [],
      "totalItems": 0
   }
}

root@7429ca762e39:/wazuh-api/test# curl -u foo:bar -k -X GET "https://127.0.0.1:55000/syscheck/000?pretty&q=file=c:\windows\system32\winrm.vbs"
{
   "error": 0,
   "data": {
      "items": [],
      "totalItems": 0
   }
}

root@7429ca762e39:/wazuh-api/test# curl -u foo:bar -k -X GET "https://127.0.0.1:55000/syscheck/000?pretty&q=file=c:\windows\system32\wbem\wmic.exe"
{
   "error": 0,
   "data": {
      "items": [],
      "totalItems": 0
   }
}

root@8e9616804657:/var/ossec/framework/python/bin# curl -u foo:bar -k -X GET "https://127.0.0.1:55000/syscheck/000?pretty&q=file=%5Bx32%5D%20HKEY_LOCAL_MACHINE%5CSystem%5CCurrentControlSet%5CServices%5Camdgpio2"
{
   "error": 0,
   "data": {
      "items": [],
      "totalItems": 0
   }
}
```

**Comment by jsanchez91** [comment]
Hi team!

We found a new error in this call, the application is complete but the answer is empty.
Examples:

**Request**
![Screenshot from 2020-06-11 10-38-29](https://user-images.githubusercontent.com/3064506/84472344-5253af80-ac87-11ea-82fb-4920ba718c69.png)
**Response**
![Screenshot from 2020-06-11 10-38-36](https://user-images.githubusercontent.com/3064506/84472341-51bb1900-ac87-11ea-9b0a-23e6e9911dd6.png)

---
**Request**
![Screenshot from 2020-06-11 10-36-56](https://user-images.githubusercontent.com/3064506/84472349-52ec4600-ac87-11ea-9812-e0194aa3467d.png)
**Response**
![Screenshot from 2020-06-11 10-37-15](https://user-images.githubusercontent.com/3064506/84472346-5253af80-ac87-11ea-9752-cf92d57ceea0.png)

---
**Request** 
![Screenshot from 2020-06-11 10-40-05](https://user-images.githubusercontent.com/3064506/84472340-51228280-ac87-11ea-9e03-15ee18f07306.png)
**Response**
![Screenshot from 2020-06-11 10-40-13](https://user-images.githubusercontent.com/3064506/84472337-5089ec00-ac87-11ea-8aae-5b70e243e70f.png)

**Comment by Selutario** [comment]
Hi team,

The problem when filtering by windows filenames should be fixed after merging the PR #5226. Here are some example requests:

```
curl -u foo:bar -k -X GET "https://127.0.0.1:55000/syscheck/001?pretty&q=file=c:\\users\\jmv74211\\desktop\\realtime\\archivo2.txt" 
{
   "error": 0,
   "data": {
      "items": [
         {
            "perm": "SYSTEM (allowed): delete|read_control|write_dac|write_owner|synchronize|read_data|write_data|append_data|read_ea|write_ea|execute|read_attributes|write_attributes, Administradores (allowed): delete|read_control|write_dac|write_owner|synchronize|read_data|write_data|append_data|read_ea|write_ea|execute|read_attributes|write_attributes, jmv74211 (allowed): delete|read_control|write_dac|write_owner|synchronize|read_data|write_data|append_data|read_ea|write_ea|execute|read_attributes|write_attributes",
            "md5": "d41d8cd98f00b204e9800998ecf8427e",
            "type": "file",
            "sha1": "da39a3ee5e6b4b0d3255bfef95601890afd80709",
            "file": "c:\\users\\jmv74211\\desktop\\realtime\\archivo2.txt",
            "sha256": "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
            "size": 0,
            "date": "2020-06-15 10:18:09",
            "uname": "Administradores",
            "uid": "S-1-5-32-544",
            "mtime": "2020-06-15 09:47:32",
            "inode": 0
         }
      ],
      "totalItems": 1
   }
}

curl -u foo:bar -k -X GET "https://127.0.0.1:55000/syscheck/001?pretty&q=file=c:\\users\\jmv74211\\desktop\\realtime\\nuevo%20documento%20de%20texto.txt" {
   "error": 0,
   "data": {
      "items": [
         {
            "size": 0,
            "file": "c:\\users\\jmv74211\\desktop\\realtime\\nuevo documento de texto.txt",
            "uid": "S-1-5-32-544",
            "inode": 0,
            "sha1": "da39a3ee5e6b4b0d3255bfef95601890afd80709",
            "sha256": "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
            "uname": "Administradores",
            "md5": "d41d8cd98f00b204e9800998ecf8427e",
            "perm": "SYSTEM (allowed): delete|read_control|write_dac|write_owner|synchronize|read_data|write_data|append_data|read_ea|write_ea|execute|read_attributes|write_attributes, Administradores (allowed): delete|read_control|write_dac|write_owner|synchronize|read_data|write_data|append_data|read_ea|write_ea|execute|read_attributes|write_attributes, jmv74211 (allowed): delete|read_control|write_dac|write_owner|synchronize|read_data|write_data|append_data|read_ea|write_ea|execute|read_attributes|write_attributes",
            "mtime": "2020-06-15 09:44:01",
            "date": "2020-06-15 10:18:09",
            "type": "file"
         }
      ],
      "totalItems": 1
   }
}

```

Best regards,
Selu.
